### PR TITLE
Filter by equipments V1.

### PR DESCRIPTION
**Pull request by mlvernay**
TODO 
- [x] CSS
- [x] search button should remain on the same line when no filters/equipments
- [x] create flag on recette and prod (see screen below)
- [x] send equipments/filters to geoJSON API
- [x] Filter ERP on equipments/filters on geoJSON API  
- [x]  Refresh map when equipments list change
- [ ] split nb stair for each part (entry, path, ...)
- [ ] put a loader on both list and map when refreshing
- [x] seems like sometimes it does not send the appropriate request when checking/unchecking a label, more visible on staging
- [x] https://sentry.incubateur.net/organizations/betagouv/issues/52976/?project=35&query=is%3Aunresolved+url%3A%22https%3A%2F%2Frecette.acceslibre.info%2Fapi%2Ferps%2Ftest-12%2F%22&referrer=issue-stream can't understand how request.GET can be `''` 
- [x] Each click on filter is generating a GET on mapicons.svg (487kb)
- [x] http://127.0.0.1:8000/recherche/?what=restaurant&where=Paris+%2875%29&lat=&lon=&code=&postcode=&search_type=in_municipality&street_name=&municipality=Paris+%2875%29 when you come from by municipality view, you have an attribute error.

image

### CODE REVIEW

**Comment by Anto59290** [comment]
Je crois que l'ajout du JS de DSFR génére des erreurs en JS, a investiguer

File: erp/provider/search.py [hunk 1]
```diff
@@ -70,3 +75,136 @@ def global_search(terms, code_insee):
...
+    if as_dict:
+        return {eq.slug: eq.name for eq in equipments}
+    return equipments
+
+
+def get_equipment_by_slug(slug: str):
+    if not slug:
+        return None
+    equipments = get_equipments()
+    for equipment in equipments:
```
Comment (mlvernay) [inline erp/provider/search.py:206]: not proud of this, and of the global structure of equipments, hard to get_by_slug, etc

File: static/js/geo.js [hunk 2]
```diff
@@ -254,6 +254,11 @@ function _getDataPromiseFromAPI(map, refreshApiUrl, apiKey, page) {
   const southWest = map.getBounds().getSouthWest()
   const northEast = map.getBounds().getNorthEast()
   const queryTerm = document.querySelector('#what-input').value
+  let equipments = document.querySelectorAll('input[name=equipments]:checked')
+  let equipmentsQuery = ''
+  equipments.forEach(function (eq) {
+    equipmentsQuery += '&equipments=' + eq.value
+  })
```
Comment (mlvernay) [inline static/js/geo.js:261]: Is there a better way to do this?

File: static/js/ui/LabelTag.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+function LabelTag(root) {
+  root.addEventListener('click', function (event) {
+    let checkbox = document.getElementById(root.getAttribute('data-for-input'))
+    checkbox.checked = !checkbox.checked
```
Comment (mlvernay) [inline static/js/ui/LabelTag.js:4]: Might be the right place to launch the map refresh

File: templates/swagger-ui.html [hunk 4]
```diff
@@ -52,9 +52,11 @@ <h1>{% translate "API acceslibre" %}</h1>
   });
         </script>
     {% else %}
-        <p>{% translate "Pour accéder à l'API, veuillez fournir une clef API valide :" %}</p>
         <form action="{% url 'set_api_key' %}" method="post">
-            <input type="text" name="api_key" />
+            <label class="fr-label" for="api_key">
+                {% translate "Pour accéder à l'API, veuillez fournir une clef API valide :" %}
+            </label>
+            <input class="fr-input" type="text" name="api_key" id="api_key" />
```
Comment (mlvernay) [inline templates/swagger-ui.html:59]: Use the DSFR here

File: tests/erp/test_search.py [hunk 5]
```diff
@@ -52,7 +52,7 @@ def test_search_pagination_performances(data, client, django_assert_max_num_quer
     )
     Accessibilite.objects.create(erp=erp, sanitaires_presence=True)
 
-    with django_assert_max_num_queries(4):
+    with django_assert_max_num_queries(5):
```
Comment (mlvernay) [inline tests/erp/test_search.py:55]: +1 for waffle flag query

File: erp/provider/search.py (hunk 1 above)
Comment (Anto59290) [reply erp/provider/search.py:206]: Why not use the as_dict param and just return equiments.get("slug") ? :thinking:

File: static/js/geo.js (hunk 2 above)
Comment (Anto59290) [reply static/js/geo.js:260]: Seems ok

**Pull request merged**